### DISCUSSION TITLE
Hardware to Provide (versioning?)

### DISCUSSION THREAD

**Discussion by hilga007** [kth5/archpower#59] [General]
Nice to see some discussions! Glad I stopped by again. I need to thin my hardware collection. @kth5 lmk if you or other contributing devs need hardware  ¯\_(ツ)_/¯..... must thin the collection. 

Using well-packaged in boxes for ATX PC cases.  modified styro. Will ship globally. Located in USA. Import restrictions may apply. 

1. PowerMac 9600 350MHz PowerPC 604ev 1MB L2 cache w/1.5GB RAM, 1TB Samsung Evo 850 mSATA SSD --> SATA Adapter --> SCSI Adapter, Radeon 9200 PCI. 
2. PowerMac 7300 200MHz PowerPC 604e 256KB L2 cache + 166MHz Pentium MMX PCI card + 1GB RAM. 
3. IntelliStation POWER 285 w/a 64-bit POWER5+ 2.1GHz + 32GB RAM. 

Have duplicate of all systems. Reasons to donate incl: 

1) EV has extra inst. vs. 604. Hope for 604-604ev optimized build, or opt. included in _powerpc_. 9600 is fastest 604x. 
2) Currently run one with daughtercard, but cannot use archlinux32 i686 packages as "multi arch" 
3) Can send IntelliStation POWER 285 also. 

EDIT: do you have separate "powerpc" builds? 

- powerpc600 (for 601-603e)
- powerpc604 (for 604-604ev) 
- powerpc700 (for powerpc g3)
- powerpc7000 (powerpc g4)
- powerpc64 for all 64-bit G5 machines and POWER "Big Endian" servers and IntelliStations. 
- powerpc64le for 64-bit POWER "Little Endian" servers and IntelliStations.

**Comment by kth5** [comment]
There are builds for powerpc64le primary, powerpc (32bit BE) and powerpc64. There simply has not been enough interest in more 32bit flavours so it is build with the G3s as a baseline while some packages such as Arcticfox and a few others do require Altivec. 

I would love to experiment with more old skool hardware for sure. Supporting OldWorld is stretching it though perhaps. PS3 support is still sort of in the making.

If you don't expect things to happen fast, I'm the last person not saving these good old 90s macs from the dumpster. But it would be cool if someone with more time on their hands can contribute. PRs welcome.

**Comment by hilga007** [comment]
Sounds good! I am trying to learn “C” at the moment. But want to see the
old Mac live again.

Would you want any packages hosted with your repo, then? Because I
definitely would want this to be beneficial to others - not just to me!

(I belonged to a late family member - so more like grave digging than
dumpster diving).

Some Googling indicates I can install Gentoo. Once Gentoo is installed,
make a partition for Arch, bootstrap a basic arch install, and then start
cloning PKGSBUILDs with the optimal MAKEPKG flags for 604ev.

Not “Easy”, but much better than writing something in C I guess :)

**Comment by kth5** [comment]
Most packages in the powerpc branch will work, base and base-devel along
with Xorg etc. The installer is built for NewWorld machines though, so your
mileage may wary.

A good start would be to look into the MacOS based OldWorld loaders such as
BootX.

One you get a Linux kernel and initramfs with drivers booted, you should
have a made more than half way. NuBus is and will probably never be
supported though, so PCI based machines are a must.

**Comment by hilga007** [comment]
Great, thanks! Have used BootX in the past with Ubuntu - back when Ubuntu PPC was a "thing". Will give it a try sometime. How would you like me to submit the instructions back to this project? A comment like this with repeatable steps? 

---- 

Then for packages -- do you want me to host my own repo, or have "somewhere" to sync back to your repo? 

---- 

Hardware - do want some shipped to you as well? 

I would like to get the IBM Intellistation with POWER5+ working as well. 32GB RAM, SSD means it is still a nice computer.

**Reply by kth5** [reply]
Sorry for the latest reply, I somehow did not follow all discussions.

We've recently had a POWER8 IBM Server added to the list of tested systems and it's a CHRP/PReP compatible beast. The `powerpc64` installer may boot on ` POWER5` as well.

I haven't figure out a proper way to integrate contributions yet since it wasn't yet necessary and my work flow is pretty particular with many shell based automations. I am open to consider integrating contribution when someone also commits for the long term and it won't add to my work load maintaining 3k+ packages.
